asv/plugins/mamba.py

```diff
@@ -155,6 +155,12 @@ def _setup(self):
                 except KeyError:
                     raise KeyError("Only pip is supported as a secondary key")
         mamba_pkgs += mamba_args
+        # Changed in v0.6.5, gh-1294
+        # previously, the user provided environment was assumed to handle the python version
+        mamba_pkgs = [
+            f"python={self._python}" if a.startswith("python") else a for a in mamba_pkgs
+        ]
+        self.context.prefix_params.target_prefix = self._path
```

Comment:
This should probably be a more strict check than `startswith` to prevent inadvertently breaking things.